show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,197 @@
+            Color.blue,
+            10f
+        )
+    }
+
+    // the actual marker graphic for the starting location
+    private val startingLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+            geometry = startingPoint
+        }
+    }
+
+    // marker graphic for the destination, location information
+    // populated on tap
+    private val endLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+        }
+    }
+
+    // the geodesic path represented as line graphic
+    private val geodesicPathGraphic by lazy {
+        val lineSymbol = SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.blue, 5f)
+        Graphic().apply {
+            symbol = lineSymbol
+        }
+    }
+
+    // the unit of distance measurement in kilometers
+    private val unitsOfMeasurement = LinearUnit(LinearUnitId.Kilometers)
+
+
+    // starting location for the distance calculation
+    private val startingPoint = Point(-73.7781, 40.6413, SpatialReference.wgs84())
+
+
+    // creates a graphic overlay to draw all graphics
+    private val graphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISImageryStandard)
+        // configure mapView assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // add all our marker graphics to the graphics overlay
+        graphicsOverlay.graphics.add(startingLocationMarkerGraphic)
+        graphicsOverlay.graphics.add(endLocationMarkerGraphic)
+        graphicsOverlay.graphics.add(geodesicPathGraphic)
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            map.load().onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> onGeoViewTapped(point) }
+                }
+            }.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map", mapView)
+            }
+        }
+    }
+
+    /**
+     * Handles the SingleTapEvent by drawing the destination location marker
+     * and a geodesic path curve using GeometryEngine.densifyGeodetic
+     * and computes the distance using
+     * GeometryEngine.lengthGeodetic
+     */
+    private fun onGeoViewTapped(point: Point) {
```

Comment:
```suggestion
    private fun displayGeodesicPath(point: Point) {
```
Recommend using function names that describe the function rather than the action. Might have to update the function comment as well.